core/src/main/java/org/elasticsearch/index/query/IndicesQueryParser.java

```diff
@@ -105,14 +93,10 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         throw new QueryParsingException(parseContext, "[indices] indices or index already specified");
                     }
                     indicesFound = true;
-                    currentIndexMatchesIndices = matchesIndices(parseContext.index().name(), parser.text());
+                    indices = new ArrayList<>();
```

Comment:
see my comment above, the current error handling is fine but I don't think we shouldn't re-initialize here